core/trino-spi/src/main/java/io/trino/spi/resourcegroups/SelectionCriteria.java

```diff
@@ -43,12 +47,39 @@ public SelectionCriteria(
         this.authenticated = authenticated;
         this.user = requireNonNull(user, "user is null");
         this.userGroups = requireNonNull(userGroups, "userGroups is null");
+        this.originalUser = requireNonNull(originalUser, "originalUser is null");
+        this.authenticatedUser = requireNonNull(authenticatedUser, "authenticatedUser is null");
         this.source = requireNonNull(source, "source is null");
         this.clientTags = Set.copyOf(requireNonNull(clientTags, "clientTags is null"));
         this.resourceEstimates = requireNonNull(resourceEstimates, "resourceEstimates is null");
         this.queryType = requireNonNull(queryType, "queryType is null");
     }
 
+    /**
+     * @deprecated Use {@link #SelectionCriteria(boolean, String, Set, String, Optional, Optional, Set, ResourceEstimates, Optional)} instead.
+     */
+    @Deprecated(since = "469", forRemoval = true)
```

Comment:
We need to remember to update this before merging.

Reply:
Good call! I updated to 473 which will be the next release, but I'll leave this thread open in case it needs another update before merge.

Reply:
Updated to 474